java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -82,4 +116,44 @@ private static boolean isBigIntegerPotentiallyBiggerThanLong(NewClassTree newCla
       return true;
     }
   }
+
+  private JavaQuickFix computeQuickFix(NewClassTree newClassTree) {
+    String message = "";
+    JavaTextEdit textEdit = null;
+    String className = QuickFixHelper.contentForTree(newClassTree.identifier(), context);
```

Comment:
if you just search for the className, you can probably just use the symbol here:
```java
String className = newClassTree.symbolType().name();
```